sys/include/senml.h

```diff
@@ -0,0 +1,158 @@
+#define SC_UPDATE_TIME   (7)    /**< Key for the update time attribute in CBOR documents   */
+#define SC_DATA_VALUE    (8)    /**< Key for the data value attribute in CBOR documents    */
+#define SC_LINK          (9)    /**< Key for the link attribute in CBOR documents          */
+
+/**
+ * @brief The different data types SenML supports
+ */
+typedef enum {
+    SENML_TYPE_UNDEF  = 0,  /**< Placeholder for when no value is provided */
+    SENML_TYPE_FLOAT  = 1,  /**< Indicates a float value                   */
+    SENML_TYPE_STRING = 2,  /**< Indicates a string value                  */
+    SENML_TYPE_BOOL   = 3,  /**< Indicates a boolean value                 */
+    SENML_TYPE_BINARY = 4   /**< Indicates a binary (data) value           */
+} senml_value_type_t;
+
+/**
+ * @brief struct that contains base information which applies to all subsequent entries
+ */
+typedef struct {
+    uint8_t             version;         /**< SenML version of this pack                            */
+    char               *base_name;       /**< Concatenated with name attribute gives sensor's UID   */
+    double              base_time;       /**< Update time values are relative to this time          */
+    char               *base_unit;       /**< When all values have the same unit, set this field    */
+    double              base_value;      /**< If not zero, add this to (float) value in each record */
+    double              base_sum;        /**< If not zero, add this to sum in each record           */
+} senml_base_info_t;
+
+/**
+ * @brief struct that contains the values of a SenML record
+ */
+typedef struct {
+    char              *name;        /**< Sensor's name, will be appended to base name if provided */
+    char              *unit;        /**< Unit of the measurement                                  */
+    char              *link;        /**< Link to additional information (must be escaped!)        */
+    double             time;        /**< Time (seconds since unix epoch) of the measurement       */
+    double             update_time; /**< Time before sensor provides an updated measurement       */
+    double             value_sum;   /**< Integrated sum of the values over time                   */
+    senml_value_type_t value_type;  /**< Indicates which type the value is of                     */
+    union {
+        double            f;  /**< A float value                              */
+        char             *s;  /**< A string value                             */
+        bool              b;  /**< A boolean value                            */
+        char             *d;  /**< A data value (binary data, base64 encoded) */
+    } value;
+} senml_record_t;
+
+/**
+ * @brief struct that holds a SenML pack (optional base info and 1..n records)
+ */
+typedef struct {
+    senml_base_info_t  *base_info;  /**< Pointer to the base info, may be NULL           */
```

Comment:
What is the reason for having `senml_base_info_t` as separate type and not merge it directly into the `senml_pack_t` structure? In the implementation it is used always through the `senml_pack_t` structure anyway... IMHO this would not only simplify the interface, but also save some de-referencing operations...